tests/ProxyManagerTest/GeneratorStrategy/FileWriterGeneratorStrategyTest.php

```diff
@@ -27,17 +30,32 @@
  *
  * @group Coverage
  * @covers \ProxyManager\GeneratorStrategy\FileWriterGeneratorStrategy
+ * @runTestsInSeparateProcesses
  *
  * Note: this test generates temporary files that are not deleted
  */
 class FileWriterGeneratorStrategyTest extends TestCase
 {
+    /** @var string */
+    private $tempDir;
+
+    protected function setUp() : void
+    {
+        $this->tempDir = sys_get_temp_dir() . '/' . self::class;
+
+        if (! is_dir($this->tempDir)) {
+            mkdir($this->tempDir);
+        }
+
+        ini_set('open_basedir', __DIR__ . '/../../..' . PATH_SEPARATOR . $this->tempDir);
```

Comment:
Is an extension in `require-dev` needed here?

Reply:
What is thia for, precisely?

Reply:
> Is an extension in require-dev needed here?

`open_basedir` is part of core.

> What is thia for, precisely?

This is for testing this scenario. `open_basedir` now restricts access to 2 paths:
- repository root
- specific directory in system temp

> If the directory does not exist or is not writable, **tempnam()** may generate a file in the system's temporary directory, and return the full path to that file, including its name. 

As briefly mentioned in [documentation of tempnam](http://php.net/manual/en/function.tempnam.php), when passed directory is not writable, this function attempts to use system temp. When system temp directory is not writable, file is not created and `false` is returned.

There is alternative approach using `putenv('TMPDIR=/not-writable-location')` as `TMPDIR` env variable can override systemp temp directory for current script. No matter which approach will be used, this line should be probably moved to `testGenerateWillFailIfTmpFileCannotBeWrittenToDisk()` to make it more obivous.